src/Nethermind/Nethermind.Evm/TransactionProcessing/TransactionProcessor.cs

```diff
@@ -152,9 +178,9 @@ private void Execute(Transaction transaction, BlockHeader block, ITxTracer txTra
                 else
                 {
                     TraceLogInvalidTx(transaction, $"SENDER_ACCOUNT_DOES_NOT_EXIST {caller}");
-                    if (isCall || gasPrice == UInt256.Zero)
+                    if (!commit || restore || gasPrice == UInt256.Zero)
                     {
-                        wasSenderAccountCreatedInsideACall = isCall;
+                        deleteCallerAccount = !commit || restore;
```

Comment:
this looks strange, very easy to forget about this pattern, will leas to mistakes

Reply:
there is test for that